src/scope.ml

```diff
@@ -118,6 +118,95 @@ type mode =
   (** Right-hand side mode. The constructor carries the environment for higher
       order variables that will be bound in the representation of the RHS. *)
 
+(* splitFunArgs t decomposes the parser term [t] (which is expected to be an application),
+  into the function symbol and a list of its actual arguments. It is implemented 
+  as a tail-recursive function. *)
+let splitFunArgs : p_term -> (p_term * (p_term list)) = fun t ->
+  let rec splitFunArgs_aux : p_term -> (p_term list) -> (p_term * (p_term list)) = fun t args ->
+    match t.elt with
+    | P_Appl(u,v) -> splitFunArgs_aux u (v::args)
+    | _           -> (t, args)
+  in
+  splitFunArgs_aux t []
+
+(* Just for debugging *)
+let boolList_to_string : bool list -> string = fun l ->
+  let rec boolList_to_string_aux : bool list -> string = fun l ->
+  match l with
+  | [] -> "] \n"
+  | h::[] -> (string_of_bool h) ^"] \n"
+  | h::t -> (string_of_bool h) ^ "; " ^ (boolList_to_string_aux t)
+  in "[" ^ boolList_to_string_aux l
+
+(** [getParamsImplicitness ss t] returns a list representing the formal parameters of 
+    a parser term [t] *)
+let getParamsImplicitness : sig_state -> p_term -> bool list = fun ss t ->
+  match t.elt with
+  | P_Prod(args, _) -> List.map (fun x -> match x with (_,_,i) -> i) args
+  | P_Abst(args,_)  -> 
+      begin
+        (* print_string "debug : P_Abst found! \n"; *)
+        List.map (fun x -> match x with (_,_,i) -> i) args
+      end
+  | P_Vari(id, imp) -> if imp=FullyExplicit then [] 
+    else
+    begin
+      (* print_string "debug : P_Vari found! \n"; *)
+      match (StrMap.find_opt (snd id.elt) ss.in_scope) with
+      | Some(res) -> let implicits = (fst res).sym_implicits in
```

Comment:
Replace res by (f,_) and (fst res) by f.

Reply:
Yes, will do, thanks.